deprecation.py

```diff
@@ -68,7 +70,10 @@ def __str__(self):
         if self.deprecated_in:
             parts["deprecated"] = " as of %s" % self.deprecated_in
         if self.removed_in:
-            parts["removed"] = " and will be removed in %s" % self.removed_in
+            if isinstance(self.removed_in, date):
+                parts["removed"] = " and will be removed on %s" % self.removed_in
```

Comment:
Just compute the preposition on the branch.

Reply:
Changed